### PR TITLE
Feature: new FFT-based spherical Bessel transform method & its unit test

### PR DESCRIPTION
**Pull request by jinzx10**

### CODE REVIEW

**Review by dyzheng — APPROVED** [review]

**Comment by dyzheng** [comment]
Wait for Doxygen format annotations.

**Comment by jinzx10** [comment]
> Wait for Doxygen format annotations.

hi @dyzheng , commit 91af356 updates the annotation format and it is now doxygen-compatible. The generated doc looks as follows:

PS: I noticed that doxygen supports various formats and the one I use is somewhat different from some existing ones... Is there an official doxyfile or something like a guideline?

image

File: source/module_base/test/spherical_bessel_transformer_test.cpp
Comment (mohanchen) [inline source/module_base/test/spherical_bessel_transformer_test.cpp:1]: I would suggest to use 'const' when possible, when you use a 'const', other developers can clearly see the property of each variable.

**Review by dyzheng — APPROVED** [review]

**Review by mohanchen — APPROVED** [review]

File: source/module_base/test/spherical_bessel_transformer_test.cpp
Comment (jinzx10) [reply source/module_base/test/spherical_bessel_transformer_test.cpp:1]: Thanks for the suggestion! const qualifiers are now used when possible.

**Pull request merged**